src/main/java/unrc/dose/Proposition.java

```diff
@@ -1,18 +1,417 @@
+     * @return the editing distance of both string
+     */
+    public static int getDistanceProposition(final Proposition p) {
+        String proposedCode = p.getSource();
+        Challenge challenge = Challenge.findById(p.get("challenge_id"));
+        String challengerCode = challenge.getString("source");
+
+        return computeLevenshteinDistance(proposedCode, challengerCode);
+    }
+
+    /**
+     * This method calculates the distance of editing between two string.
+     * @param str1 represents the original string
+     * @param str2 represents the modified string
+     * @return the editing distance of both string
+     */
+    public static int computeLevenshteinDistance(
+        final String str1,
+        final String str2) {
+        return computeLevenshteinDistance(
+            str1.toCharArray(),
+            str2.toCharArray()
+            );
+    }
+
+    /**
+     * Get the minimum between three characters.
+     * @param a represent a integer value
+     * @param b represent a integer value
+     * @param c represent a integer value
+     * @return represent the minimum value.
+     */
+    private static int minimum(final int a, final int b, final int c) {
+        if (a <= b && a <= c) {
+            return a;
+        }
+        if (b <= a && b <= c) {
+            return b;
+        }
+        return c;
+    }
+
+    /**
+     * This method calculates the distance of editing
+     * between two character arrangements.
+     * @param str1 represents the original arrangement
+     * @param str2 represents the modified arrangement
+     * @return the editing distance of both arrangements
+     */
+    private static int computeLevenshteinDistance(
+        final char[] str1,
+        final char[] str2) {
+        int[][] distance = new int[str1.length + 1][str2.length + 1];
+
+        for (int i = 0; i <= str1.length; i++) {
+            distance[i][0] = i;
+        }
+        for (int j = 0; j <= str2.length; j++) {
+            distance[0][j] = j;
+        }
+        for (int i = 1; i <= str1.length; i++) {
+            for (int j = 1; j <= str2.length; j++) {
+                distance[i][j] = minimum(distance[i - 1][j] + 1,
+                        distance[i][j - 1] + 1, distance[i - 1][j - 1]
+                                + ((str1[i - 1] == str2[j - 1]) ? 0 : 1));
+            }
+        }
+        return distance[str1.length][str2.length];
+    }
+
+    /**
+     * Generate a .java file with the code of a challenge.
+     * @param source text with code java
+     */
+    public static void generateFileJava(final String source) {
+
+        File f;
+        f = new File("Source.java");
+
+        try {
+            FileWriter w = new FileWriter(f);
+            BufferedWriter bw = new BufferedWriter(w);
+            PrintWriter wr = new PrintWriter(bw);
+            wr.write(source);
+            wr.close();
+            bw.close();
+        }
+        catch (IOException e) {
+        };
+    }
```

Comment:
A class file must have the same name as the class defined inside. Using `Source.java` will probably fail.
There is no check for the argument and no check for the resulting file.

This method hides the exception without even logging it